LibSWBF2/Chunks/LVL/lght/vec3.DATA.cpp

```diff
@@ -0,0 +1,31 @@
+#include "stdafx.h"
+#include "vec3.DATA.h"
+#include "InternalHelpers.h"
+#include "FileReader.h"
+#include "Types/Enums.h"
+#include "Logging/Logger.h"
+
+
+namespace LibSWBF2::Chunks::LVL::lght
+{
+	void DATA_VEC3::RefreshSize()
+	{
+		THROW("Not implemented!");
+	}
+
+	void DATA_VEC3::WriteToStream(FileWriter& stream)
+	{
+		THROW("Not implemented!");
+	}
+
+	void DATA_VEC3::ReadFromStream(FileReader& stream)
+	{
+		BaseChunk::ReadFromStream(stream);
+        Check(stream);
+
+        stream.SkipBytes(5);
```

Comment:
First 5 bytes contain a 4 byte ID for the (vector) data (ie locations, rotation) and the next byte holds the number of floats in the vector. This could be rewritten to return a new derivation of DATA based on the 4 byte ID instead of order, but from all lvls I've looked through the 4 common fields (rotation, location, light type, color) always occur in the same order. I'll probably change to an ID based approach when dealing with the specific light fields + global lighting data...